### PR TITLE
Support configuration of maxCapacityPerShard when reading from Kinesis with EFO

### LINKED ISSUE #26257: [Task]: Allow configuring `withMaxCapacityPerShard` for KinesisIO with EFO
### What needs to happen?

Current version of `EFOShardSubscriber` does not take into account `KinesisIO.Read#withMaxCapacityPerShard(Integer)` config and has hard-coded in-flight limit: https://github.com/apache/beam/blob/cd0f44b3751c9d4b583b2a3e2e1d10aeb75695b9/sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/EFOShardSubscriber.java#L59

Default for `maxCapacityPerShard` should be set to `null` and each type of consumer (non-EFO or EFO) should have their own defaults (10K is too large for EFO).

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by mosche**
Support configuration of maxCapacityPerShard when reading from Kinesis with EFO
(closes #26257)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by mosche** [comment]
R: @psolomin 
R: @aromanenko-dev

File: sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisIO.java [hunk 1]
```diff
@@ -615,7 +614,15 @@ public Read withCustomRateLimitPolicy(RateLimitPolicyFactory rateLimitPolicyFact
...
+    /**
+     * Specifies the maximum number of messages per one shard.
+     *
+     * <p>Note: When using consumers with dedicated throughput (Enhanced Fan-Out), this capacity
+     * corresponds to the number of in-flight shard events which itself can contain multiple,
+     * potentially even aggregated records.
+     *
+     * @see {@link #withConsumerArn(String)}
+     */
     public Read withMaxCapacityPerShard(Integer maxCapacity) {
```
Comment (psolomin) [inline sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisIO.java:626]: Looks like the line 228 should be updated too:

```
 * <p>When EFO is enabled, the following configurations are ignored:
 *
 * <ul>
 *   <li>{@link Read#withMaxCapacityPerShard(Integer)}   <--- drop this?
 *   <li>{@link Read#withRequestRecordsLimit(int)}
```

File: sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/EFOShardSubscribersPool.java [hunk 2]
```diff
@@ -470,6 +472,10 @@ String getPoolId() {
     return poolId;
   }
 
+  public int getMaxCapacityPerShard() {
```
Comment (psolomin) [inline sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/EFOShardSubscribersPool.java:475]:
```suggestion
  int getMaxCapacityPerShard() {
```

**Review by psolomin — COMMENTED** [review]
Thanks 👍 looks good overall.

I left a couple of small suggestions.

**Comment by mosche** [comment]
Thanks @psolomin, good catch! I've addressed your comments.

**Review by psolomin — APPROVED** [review]
👍

**Pull request merged**